core/container.go

```diff
@@ -1094,7 +1093,7 @@ func (container *Container) WithExec(ctx context.Context, gw bkgw.Client, defaul
 		return nil, fmt.Errorf("marshal: %w", err)
 	}
 	hostname := hostHash(digest)
-	payload.Hostname = fmt.Sprintf("%s.%s", hostname, network.DNSDomain)
+	payload.Hostname = hostname
```

Comment:
We actually just don't need to use the domain at all now; it's not even configured in the `/etc/resolv.conf` we set up. Now that everything is routed directly to `dnsmasq` with no other servers listed in `resolv.conf` there should never be any ambiguity.

There might be a small benefit in including it and installing our domain into the `search` list, because it would allow us to skip the domain searching routine if the user's original `/etc/resolv.conf` has them configured (which we preserve), but even that benefit only applies as long as they don't also have `options ndots:N` set ([k8s defaults to `ndots:5`](https://pracucci.com/kubernetes-dns-resolution-ndots-options-and-why-it-may-affect-application-performances.html)). So it's a bit of a wash.

The benefit of _not_ including it, on the other hand, is that we can keep `core/` decoupled from the engine configuration, which definitely feels a lot nicer.

Reply:
Oh okay interesting, I got confused when I saw we still set something in the dnsmasq template related to the domain, but that all makes sense, thanks for the explanation!